src/main/kotlin/stringcalculator/domain/customparser/CustomNumbersStringParser.kt

```diff
@@ -0,0 +1,48 @@
+package stringcalculator.domain.customparser
+
+class CustomNumbersStringParser(stringExpression: String, separators: ParserSeparators) {
+    val parsedNumbers: List<Int>
+
+    init {
+        validateSeparatorsEmpty(separators)
+        validateExpressionIncludeOtherStrings(stringExpression, separators)
+
+        parsedNumbers = splitBySeparators(stringExpression, separators)
+        validateParsedNumberForIncludeNegativeNumber(parsedNumbers)
+    }
+
+    private fun splitBySeparators(
+        stringExpression: String,
+        separators: ParserSeparators
+    ) = stringExpression.split(Regex(separators.getSeparatorsString().toString())).map { it.toInt() }
+
+    private fun validateSeparatorsEmpty(separators: ParserSeparators) {
+        require(separators.getSize() > 0) { ERROR_MESSAGE_SEPARATOR_ZERO }
+    }
+
+    companion object {
+        private const val ERROR_MESSAGE_SEPARATOR_ZERO = "숫자를 추출하기 위한 구분자가 없습니다"
+
+        private fun getErrorMessageIncludeNegativeNumber(negativeNumber: Int): String {
+            return "음수($negativeNumber)은 입력할수 없습니다"
+        }
+
+        fun validateParsedNumberForIncludeNegativeNumber(parsedNumbers: List<Int>) {
+            parsedNumbers.forEach { require(it >= 0) { getErrorMessageIncludeNegativeNumber(it) } }
+        }
```

Comment:
private 메서드는 사용되는 메서드 바로 아래쪽에 위치하는것이 읽기 더 좋습니다!

```suggestion
        fun validateParsedNumberForIncludeNegativeNumber(parsedNumbers: List<Int>) {
            parsedNumbers.forEach { require(it >= 0) { getErrorMessageIncludeNegativeNumber(it) } }
        }
        
        private fun getErrorMessageIncludeNegativeNumber(negativeNumber: Int): String {
            return "음수($negativeNumber)은 입력할수 없습니다"
        }
```